### ISSUE TITLE
Copy ContentType API

### ISSUE THREAD

**Issue by swicken-dotcms** [dotCMS/core#21697]
**Is your feature request related to a problem? Please describe.**
There are circumstances that might require the creation of many content types that share the same base fields. Creating the same fields each time adds tedium and could result in user error.

**Describe the solution you'd like**
There are multiple potential ways to implement this, however after some discussion we think the best solution would be to allow a Content Type to copy its initial fields from another Content Type. This would allow users to create bootstrap Content Types for a lot of default fields, or extend existing already created Content Types.

**Comment by john-thomas-dotcms** [comment]
Here's another potential solution that I think would also work, and might offer some advantages:  Allow users to “embed” one content type in another.

So, you could create a content type A with the common fields - including the velocity var names and layout those fields should always have - and then embed that entire content type into content type B.  The embedding could be done multiple ways, though I think a new field type might be the simplest.

Some potential advantages:
1. Changing the base (embedded) content type could potentially automatically change all the other content types that embed it.
    * If you wanted to change the shared fields with a solution that just copies the field, you'd have to manually change all content types that copied those fields.
2, The embedded fields could be visually displayed as being grouped together, rather than showing as individual separate fields.
3. The fields could be moved around within the "outer" content type as a block (retaining their layout in the process).
4. The data for the fields could potentially be stored in the table as being content of the inner/embedded content type, instead of being content of the outer content type.  (Possibly linked via some kind of hidden relationship).
    * A single search, then, could be made against all content which includes fields from the embedded type.
    * Operations such as delete field could then also be done on the source content type, affecting all content types that use it - with little to no change in existing logic for these operations.
    * These embedded fields could then also potentially be accessed via a single object (via Velocity, GraphQL query results, etc.) rather than as entirely separate fields.  Meaning $content.embeddedGroup.varA instead of just $content.varA.
        * This, in turn, would allow embedding of fields even if the var names of those fields conflict with var names of fields already in the content type the fields are embedded into.

There are some other potential advantages this could offer, though I think the use cases are alot less common than the ones I mentioned above.  In any case, another alternative to consider.

**Comment by erickgonzalez** [comment]
Endpoint: `/api/v1/contenttype/{{VAR_NAME_CONTENTTYPE_BASE}}/_copy` and the Body (JSON) can receive this:
```
name REQUIRED
variable OPTIONAL
folder OPTIONAL
host OPTIONAL
icon OPTIONAL
```

e.g
```
/api/v1/contenttype/fileAsset/_copy

{
    "name": "File Asset Copy"
}
```

**Comment by bryanboza** [comment]
Tested and now we the endpoint is working and we are able to clone content types, but here some improvements

- If you don't send the name, we are returning a 500 error and `null` message in the response. Also multiple lines of error in the log, we need catch this exception and be clear with the error at log level. 
Here the log: https://gist.github.com/bryanboza/e9d923f19b9e14ab2c7c58114b439397

- If you have relationship fields in the content type, we are altering the order of the fields and leave those ones at the end of the content type. We should create an exactly copy of the content type 

- We are not respecting the "host" parameter, always the copy set the same of the original one.

**Comment by bryanboza** [comment]
This card will need to doc: 
Here the postman doc examples: https://t.ly/Ian4

**Comment by jdotcms** [comment]
This is the new [PR](https://github.com/dotCMS/core/pull/21900/files) 
The point:
If you have relationship fields in the content type, we are altering the order of the fields and leave those ones at the end of the content type. We should create an exactly copy of the content type

Couldn't be reproduce the screenshot you can see how the copied content type looks such as the original content type
<img width="1244" alt="Screen Shot 2022-03-25 at 1 11 54 PM" src="https://user-images.githubusercontent.com/19578350/160205383-cb115153-3829-464a-bef3-dff38ccc041c.png">
<img width="1278" alt="Screen Shot 2022-03-25 at 1 12 01 PM" src="https://user-images.githubusercontent.com/19578350/160205366-bc8eedb3-45fd-4d2b-b95f-e7b50be8d4fa.png">

**Comment by jdotcms** [comment]
After a call with @bryanboza I am able to see this behavior
<img width="1298" alt="Screen Shot 2022-04-06 at 3 00 25 PM" src="https://user-images.githubusercontent.com/19578350/162070481-68583754-8a76-417b-ab2b-4b7b8c3b324a.png">

**Comment by bryanboza** [comment]
Also, we need to test this using MSSQL as db, I'm getting this error when I try to copy the `fileAsset` content type 
![image](https://user-images.githubusercontent.com/2641437/162070798-b02a05d0-0033-42ae-bbc3-3ca042b01242.png)

Log Error: https://gist.github.com/bryanboza/240664e8451cd03e5a3929de9caeb4c0

**Comment by jdotcms** [comment]
[PR](https://github.com/dotCMS/core/pull/22000/files)

**Comment by fmontes** [comment]
As per @oidacra order is fine now.

**Comment by oidacra** [comment]
Verified using docker image `22.04_4210df8_SNAPSHOT`

**Comment by bryanboza** [comment]
Fixed, tested on release-22.04 // Docker // FF
